dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx

```diff
@@ -86,21 +86,34 @@ class WorkloadItem extends React.Component<IResourceItemProps> {
 
   private renderResource = (r: IResource | ISecret) => {
     const plainResource = r as IResource;
+    const type = this.typeFromSelfLink(r.metadata.selfLink);
     // resource kind may not be available for Lists
-    if (r.metadata.selfLink.match("deployments")) {
-      return <DeploymentItemRow resource={plainResource} />;
-    } else if (r.metadata.selfLink.match("statefulsets")) {
-      return <StatefulSetItemRow resource={plainResource} />;
-    } else if (r.metadata.selfLink.match("daemonsets")) {
-      return <DaemonSetItemRow resource={plainResource} />;
-    } else if (r.metadata.selfLink.match("services")) {
-      return <ServiceItem resource={plainResource} />;
-    } else if (r.metadata.selfLink.match("secrets")) {
-      return <SecretItem resource={r as ISecret} />;
-    } else {
-      return <OtherResourceItem resource={plainResource} />;
+    switch (type) {
+      case "deployments":
+        return <DeploymentItemRow resource={plainResource} />;
+      case "statefulsets":
+        return <StatefulSetItemRow resource={plainResource} />;
+      case "daemonsets":
+        return <DaemonSetItemRow resource={plainResource} />;
+      case "services":
+        return <ServiceItem resource={plainResource} />;
+      case "secrets":
+        return <SecretItem resource={r as ISecret} />;
+      default:
+        return <OtherResourceItem resource={plainResource} />;
     }
   };
+
+  private typeFromSelfLink(link: string) {
+    const parsedLink = link.split("/");
+    if (parsedLink.length < 2) {
+      // Unknown selflink
+      return "";
+    }
+    // For a single resource, the type is the second-to-last item
+    // e.g. /api/v1/namespaces/default/pods/foo
+    return parsedLink[parsedLink.length - 2];
```

Comment:
Minor point, but it seems like `typeFromSelfLink` is also where the switch should belong: it's the function which encapsulates returning the resource type (ie. not the string, but the object). Up to you.

Reply:
the goal of this function was to be a bit more self-readable, if it's not I will just use `renderResource` above